conferences/presentations/UKRI_cloud_2019-02-12/UKRI_cloud_demo_2019-02-12.md

```diff
@@ -0,0 +1,129 @@
+
+<html><img src="figure2.png" alt="figure2" height="400" width="320"></html>
+
+* This is a figure published in [my first journal paper](https://arxiv.org/pdf/1706.04802.pdf)
+* It describes the evolution of the spin of a neutron star (middle panel) and the mass of its accretion disc (top panel) as it was being fed by fallback accretion
+* The bottom panel is the property we were interested in which is when a derived parameter would cross a specific threshold (the dashed line)
+
+**What my research was about isn't necessarily important.
+What is important is whether other scientists in my field could verify my work.**
+
+<html><img src="fig2_creation.gif" alt="fig2_creation"></html>
+
+This is a GIF of my PhD laptop producing the figure which turned out to **not** be reproducible for a number of reasons:
+* Was not version controlled
+* Computing environment(s) was not documented
+* Computing environment no longer exists - the laptop has been returned and wiped :scream:
+
+## Binder to the Rescue!
+
+With a little bit of work, I've managed to reproduce my figure using Binder (https://mybinder.org).
+
+* My code is in a public GitHub repo - now version controlled :ballot_box_with_check:
+* The computing environment has been documented in an `environment.yml` file :ballot_box_with_check: _(other config file types are available)_
+
+My PhD repo: https://bit.ly/2DacNJQ
+
+Link to back-up GIF: https://bit.ly/2GjmCt9
+
+## What is Binder doing?
+
+<html><img src="binder_demo.jpg" alt="mybinder" height="420" width="540"></html>
+
+Courtesy of [Juliette Belin](https://twitter.com/JulietteTaka/status/1082735653929000960)
+
+Read the docs on making your own repo Binder-ready here: https://bit.ly/2t7M9wx
+
+## Limitations with the public Binder instance
+
+* Only works for **public** repos, cannot host private code or sensitive data
+* Large datasets are discouraged
+* Computing resources are minimal
+
+## Solution: BinderHub 4 U
+
+<html><img src="binder_demo2.png" alt="yourbinder" height="420" width="580"></html>
+
+* The host institution/organisation/RSE group can **choose** whether to make repos public or private
+* This is on-going project at the Turing Institute
+
+**Some useful links:**
+* [Zero-to-JupyterHub](https://zero-to-jupyterhub.readthedocs.io/en/latest/)
+* [Zero-to-JupyterHub with Kubernetes](https://zero-to-jupyterhub.readthedocs.io/en/latest/index.html)
+* [Step Zero: Kubernetes on Microsoft Azure](https://zero-to-jupyterhub.readthedocs.io/en/latest/microsoft/step-zero-azure.html)
+* [repo2docker](https://repo2docker.readthedocs.io/en/latest/?badge=latest)
+
+## Why tracking Dependencies is Important (demo)
+
+Version updates to software packages could cause fundamental changes to your code that do not raise a fatal error, and so will pass without you realising.
+
+Here's a little demo repo to highlight this: https://bit.ly/2DfpZNo
```

Comment:
This repo needs to be merged into https://github.com/binder-examples

Reply:
(and can then have a line similar to your PhD line above:

```suggestion
Here's a little demo repo to highlight this: https://github.com/binder-examples/different-versions-matplotlib, binder: bit.ly....
```

Reply:
Added my own version of this suggestion. I asked the question in the binder gitter about how I submit this to binder-examples as I don't know how to make a PR to an organisation. I doubt it'll be done by tomorrow though.